packages/core-mobile/app/new/features/swap/contexts/SwapContext.tsx

```diff
@@ -70,7 +73,10 @@ export const SwapContextProvider = ({
   const [swapStatus, setSwapStatus] = useState<SwapStatus>('Idle')
   const [amount, setAmount] = useState<bigint>()
   const [isFetchingQuote, setIsFetchingQuote] = useState(false)
-  const [quote, setQuote] = useState<SwapQuote | undefined>()
+  const [quote, setQuote] = useState<SwapQuote | undefined>() // legacy
+  const [selectedQuote, setSelectedQuote] = useSelectedSwapRate()
+  const [bestQuote, setBestQuote] = useBestRate()
+  const [allQuotes, setAllQuotes] = useAllRates()
```

Comment:
I would also normalize the quote object so that all providers can just return the same object. it can be something like
```
type NormalizedQuoteResult = {
  allQuotes: SwapQuote[]
  selectedQuote: SwapQuote
  providerMetadata?: Record<string, unknown>
}

MarkrProvider - getQuote:
  {
    allQuotes: bestRate.routes,
    selectedQuote: bestRate.routes[0], // or use a helper to rank/sort
    providerMetadata: { destAmount }
  }

ParaswapProvider - getQuote:
  {
    allQuotes: [optimalRate],
    selectedQuote: optimalRate,
    providerMetadata: { destAmount }
  }
```
then in the UI, you can allow users to select if there are more than 1 quote